### PR TITLE
Add GetCollidingEntities to the API

### LINKED ISSUE #1036: CollidableComponent should be able to retrieve entities it is colliding with
Right now, `TryCollision` only returns a boolean value, not the entities that are causing the collision. It would be nice to have a `GetCollidingEntities` function that would retrieve all entities that are colliding with the object.

### PR DESCRIPTION
**Pull request by ComicIronic**
Fixes #1036.

### CODE REVIEW

File: Robust.Shared/GameObjects/Components/Collidable/CollidableComponent.cs [hunk 1]
```diff
@@ -226,6 +226,15 @@ public bool TryCollision(Vector2 offset, bool bump = false)
             return _physicsManager.TryCollide(Owner, offset, bump);
         }
 
+        /// <inheritdoc />
+        public IEnumerable<IEntity> GetCollidingEntities(Vector2 offset, bool bump = false)
+        {
+            if (!_collisionEnabled || CollisionMask == 0x0)
+                return new IEntity[] { };
+
+            return null;
```
Comment (ComicIronic) [inline Robust.Shared/GameObjects/Components/Collidable/CollidableComponent.cs]: Hwoops.

**Comment by snowsignal** [comment]
Thanks for doing this, but this has already been added as part of my physics rewrite PR #1037. I should have added #1036 to the list of issues that the PR was going to fix.

File: Robust.Shared/Physics/PhysicsManager.cs [hunk 2]
```diff
@@ -109,21 +120,21 @@ public bool TryCollide(IEntity entity, Vector2 offset, bool bump = true)
...
-                    continue;
+                    otherCollidable.Bumped(entity);
                 }
-
-                otherCollidable.Bumped(entity);
-                collidedwith.Add(otherCollidable.Owner);
             }
 
-            collidable.Bump(collidedwith);
+            if (bump)
```
Comment (ComicIronic) [inline Robust.Shared/Physics/PhysicsManager.cs:131]: @InquisitivePenguin Does your PR also fix this missing check in some way?

Comment (snowsignal) [reply Robust.Shared/Physics/PhysicsManager.cs:131]: My PR removes the whole 'bump' concept entirely, so... technically yes?

**Comment by gradientvera** [comment]
Mind if I close this in favor of #1037 then? @ComicIronic @InquisitivePenguin

**Comment by ComicIronic** [comment]
Sounds good.

**Pull request closed without merging**